aten/src/ATen/detail/XLAHooksInterface.cpp

```diff
@@ -0,0 +1,22 @@
+#include <ATen/detail/XLAHooksInterface.h>
+
+#include <atomic>
+
+namespace at {
+namespace detail {
+namespace {
+
+std::atomic<XLAHooksInterface*> g_xla_hooks(nullptr);
+
+}  // namespace
+
+const XLAHooksInterface* getXLAHooks() {
```

Comment:
as above, can we follow the HIP pattern?

Reply:
Tried to explain above, so maybe I can close this one and keep single thread.